src/bin/main.rs

```diff
@@ -30,11 +30,19 @@ fn main() -> Result<(), Error> {
     // Start dhall configuration loader
     loading_rt.spawn(compiler_executor(config_receiver, state.clone()));
 
-    // Load configuration files
-    loading_rt.spawn(load_configuration_files(
-        cli_args.configuration_files.into_iter(),
-        config_sender,
-    ));
+    if cli_args.wait_configuration_load {
+        // Load configuration files synchronously
+        info!("Wait for configuration loading");
+        for config in cli_args.configuration_files {
```

Comment:
You are loading configuration on the main thread, is that what you want?

Reply:
Yes it's the goal of this flag
Load the configruation on main thread to block everything is compile and loaded before starting the wb server